.github/actions/install-aiida-core/action.yml

```diff
@@ -29,19 +31,18 @@ runs:
       python-version: ${{ inputs.python-version }}
 
   - name: Set up uv
-    uses: astral-sh/setup-uv@v4
+    uses: astral-sh/setup-uv@v5
     with:
-      version: 0.5.6
+      version: 0.5.x
+      python-version: ${{ inputs.python-version }}
 
   - name: Install dependencies from uv lock
     if: ${{ inputs.from-lock == 'true' }}
     # NOTE: We're asserting that the lockfile is up to date
-    # NOTE2: 'pre-commit' extra recursively contains other extras
-    # needed to run the tests.
-    run: uv sync --locked --extra pre-commit
+    run: uv sync --locked ${{ inputs.extras && format('--extra {0}', inputs.extras) || '' }}
     shell: bash
 
   - name: Install aiida-core
     if: ${{ inputs.from-lock != 'true' }}
-    run: uv pip install --system -e .${{ inputs.extras }}
```

Comment:
Since now `setup-uv` action activates the virtual environment automatically, we no longer need to install into system python.